src/main/webapp/scripts/group-details-script.js

```diff
@@ -0,0 +1,314 @@
+    }
+    // Find out differences between this group's settings and default public settings
+    var diffs = diff(group, publicSettings);
+    console.log(diffs)
+    if (Object.keys(diffs).length == 0) {
+        return "Public";
+    } else if (Object.keys(diff(teamDiffs, diffs)).length == 0) {
+        return "Team";
+    } else if (Object.keys(diff(announcementDiffs, diffs)).length == 0) {
+        return "Announcement Only";
+    } else if (Object.keys(diff(restrictedDiffs, diffs)).length == 0) {
+        return "Restricted";
+    } else {
+        return "Custom";
+    }
+}
+
+/** Returns an object containing the key and value pairs for each difference between two objects
+  * obj1: object you are comparing to default
+  * obj2: default object
+ */
+function diff(obj1, obj2) {
+    var diffs = {};
+    for (key in obj2) {
+        if (obj2.hasOwnProperty(key)) {
+            if (obj1[key] !== obj2[key]) {
+                diffs[key] = obj1[key];
+            }
+        }
+    }
+    return diffs;
+}
+
+/** Function called when user clicks to view access and membership settings */
+function viewSettings() {
+    var settingsChart = document.getElementById("group-settings-chart");
+    if (settingsChart.classList.contains("hidden")) {
+        settingsChart.classList.remove("hidden")
+    } else {
+        settingsChart.classList.add("hidden")
+    }
+}
+
+/** Fills in the checkbox table for access and membership settings */
+function setAccessMembershipSettingsTable(json) {
+    // create the access and membership settings table chart
+    var accessSettingsCheckboxes = [
+            [document.getElementById("contact-owners-group-owners"),
+            document.getElementById("contact-owners-group-managers"),
+            document.getElementById("contact-owners-group-members"),
+            document.getElementById("contact-owners-entire-organization"),
+            document.getElementById("contact-owners-external"),],
+            [document.getElementById("view-members-group-owners"),
+            document.getElementById("view-members-group-managers"),
+            document.getElementById("view-members-group-members"),
+            document.getElementById("view-members-entire-organization"),],
+            [document.getElementById("view-topics-group-owners"),
+            document.getElementById("view-topics-group-managers"),
+            document.getElementById("view-topics-group-members"),
+            document.getElementById("view-topics-entire-organization"),
+            document.getElementById("view-topics-external"),],
+            [document.getElementById("publish-posts-group-owners"),
+            document.getElementById("publish-posts-group-managers"),
+            document.getElementById("publish-posts-group-members"),
+            document.getElementById("publish-posts-entire-organization"),
+            document.getElementById("publish-posts-external"),],
+        ]
+    accessSettingsChecked = [
```

Comment:
I would refactor this variable so that the boolean values correspond to what they enable. As of now its just a list of booleans that enable or disable the settings from accessSettingsCheckboxes

Reply:
yes doing that right now!

Reply:
Something like a dictionary of setting: boolean might make it easier to read